lib/core/src/Cardano/Wallet.hs

```diff
@@ -3436,6 +3433,7 @@ data ErrBalanceTx
     | ErrBalanceTxAssignRedeemers ErrAssignRedeemers
     | ErrBalanceTxInternalError ErrBalanceTxInternalError
     | ErrBalanceTxZeroAdaOutput
+    | ErrByronTxNotSupported
```

Comment:
Perhaps this could be `ErrTxEraNotSupported`, with a parameter to express the era?

Mentioning Byron here seems a bit "specific" for the generality of the `ErrBalanceTx` type.

Reply:
Perhaps you could have:

```suggestion
    | ErrEraNotSupported EraNotSupported

data EraNotSupported = ErrByronTxNotSupported
```

Reply:
I really _do_ like avoiding unnecessary nesting. Can revisit in the future if needed.